server/channels/app/migrations.go

```diff
@@ -585,12 +585,21 @@ func (s *Server) doCloudS3PathMigrations() {
 
 	// If there is a job already pending, no need to schedule again.
 	// This is possible if the pod was rolled over.
-	jobs, err := s.Store().Job().GetAllByTypeAndStatus(model.JobTypeS3PathMigration, model.JobStatusPending)
+	cnt, err := s.Store().Job().GetCountByStatusAndType(model.JobStatusPending, model.JobTypeS3PathMigration)
```

Comment:
Note that there is a method `GetNewestJobByStatusesAndType` which accepts multiple statuses, but I chose two separate calls for simplicity. In the other one, I would have had to distinguish between normal error and no rows error. This style keeps things simpler IMO.

Reply:
If the race occurs from two pods, what prevents the race from just moving to line 604? It feels like a more robust solution to jobs not running concurrently would be to have `CreateJob` fail in a way that the loser can gracefully decline. Do we have constraints on the table that only allow one job in a given state?

Reply:
> Do we have constraints on the table that only allow one job in a given state?

No we don't. But we can create another method to allow creation of only one pending/in-progress job.